src/pages/home/report/withReportAndReportActionOrNotFound.tsx

```diff
@@ -103,7 +102,7 @@ export default function <TProps extends WithReportAndReportActionOrNotFoundProps
 
     WithReportOrNotFound.displayName = `withReportOrNotFound(${getComponentDisplayName(WrappedComponent)})`;
 
-    return compose(
```

Comment:
In most places it won't make any difference, but we have to take into consideration the order of these functions:
```ts
function a(x: string) {
    return x + 'a';
}

function b(x: string) {
    return x + 'b';
}

console.log(compose(a, b)('c')); // cba
console.log(a(b('c'))); // cba
console.log(b(a('c'))); // cab
```

If I'm not wrong you should do it the other way around here and all other places in your PR.

Reply:
Here it should be:
```ts
withOnyx(withWindowDimensions(...
```

Reply:
I see... okay will fix that

Reply:
I tried to change it but here and in all other places i got a typescript error in most of them, so i am not sure if this is the right way to do it, could you please let me know if this is the correct change or not ? @barros001  thanks


```js

 return withOnyx<TProps & RefAttributes<TRef>, OnyxProps>({
        report: {
            key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT}${route.params.reportID}`,
        },
        parentReport: {
            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT}${report ? report.parentReportID : '0'}`,
        },
        reportMetadata: {
            key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT_METADATA}${route.params.reportID}`,
        },
        isLoadingReportData: {
            key: ONYXKEYS.IS_LOADING_REPORT_DATA,
        },
        betas: {
            key: ONYXKEYS.BETAS,
        },
        policies: {
            key: ONYXKEYS.COLLECTION.POLICY,
        },
        reportActions: {
            key: ({route}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${route.params.reportID}`,
            canEvict: false,
        },
        parentReportAction: {
            key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report ? report.parentReportID : 0}`,
            selector: (parentReportActions: OnyxEntry<OnyxTypes.ReportActions>, props: WithOnyxInstanceState<OnyxProps>): OnyxEntry<OnyxTypes.ReportAction> => {
                const parentReportActionID = props?.report?.parentReportActionID;
                if (!parentReportActionID) {
                    return null;
                }
                return parentReportActions?.[parentReportActionID] ?? null;
            },
            canEvict: false,
        },
    })(withWindowDimensions(React.forwardRef(WithReportOrNotFound)));
}
```

also [here](https://github.com/Expensify/App/blob/4400aca0b2948f8a2945e16335975152851df5b6/src/components/LocaleContextProvider.tsx#L127-L133), i changed accordingly and still get ts errors:

```js
const Provider = withOnyx<LocaleContextProviderProps, LocaleContextProviderOnyxProps>({
    preferredLocale: {
        key: ONYXKEYS.NVP_PREFERRED_LOCALE,
        selector: (preferredLocale) => preferredLocale,
    },
})(withCurrentUserPersonalDetails(LocaleContextProvider));
```

Reply:
I think blazej meant [here](https://github.com/Expensify/App/issues/39117#issuecomment-2102293408) to keep the hook to another migration iteration. 

cc @blazejkustra

Reply:
These types are quite complicated and I don't want to waste too much time to fix these issues, I see two approaches:
1. Use a hook to avoid unnecessary errors
2. Leave the order of the HOCs as you implemented it initially (only for places where TS is complaining) 
```ts
withWindowDimensions(withOnyx(...
```

Reply:
If you want to dig deeper into this issue @shubham1206agra I think the problem lies in `withReportAndReportActionOrNotFound` return type, it should be something like this: `ComponentType<Omit<TProps & RefAttributes<TRef>, keyof OnyxProps | keyof WindowDimensionsProps>>`

Reply:
>  Use a hook to avoid unnecessary errors <br>  Leave the order of the HOCs as you implemented it initially (only for places where TS is complaining)
withWindowDimensions(withOnyx(...

which approach of the two should i go with @shubham1206agra ?

Reply:
kindly reminder